deploy.py

```diff
@@ -52,8 +54,13 @@ def run_command(command: str) -> str:
     return "".join(output)
 
 
-def extract_contract_address(contract_name: str, chain_id: str) -> str:
-    path = f"broadcast/{contract_name}.s.sol/{chain_id}/run-latest.json"
+def extract_contract_address(contract_name: str, chain_id: str, is_router: bool) -> str:
+    if is_router:
+        function_signature = ROUTER_SCRIPT_RUN_SIGNATURE[2:]  # do [2:] to remove 0x
+    else:
+        function_signature = "run"
+
+    path = f"broadcast/{contract_name}.s.sol/{chain_id}/{function_signature}-latest.json"
```

Comment:
why is this necessary? isn't `contract_name` already enough?

Reply:
Unfortunately no :( Foundry identifies a script run by script name + function that was called inside the script. And since for routers the `run` function contains an argument, it has a different signature that the `run` without args.

Reply:
Got it. Seems like the better approach then would be to get the signature dynamically: 

- read files in `broadcast/{contract_name}.s.sol/{chain_id}/` and set path to the one that contains `latest`

That way we don't have to `ROUTER_SCRIPT_RUN_SIGNATURE` everytime the input params change